tests/test_login_functionality.py

```diff
@@ -20,6 +20,7 @@ def test_login_functionality(self):
 
         username_field = driver.find_element(By.XPATH, '//input[@data-test="username"]')
         username_field.send_keys(USERNAME)
+        # TODO: add step for checking username field value
         assert username_field.is_displayed(), "The username is not visible."
 
         password_field = driver.find_element(By.XPATH, '//input[@data-test="password"]')
```

Comment:
@Sv-24 wrap code for checking url to method verify_current_url, where you send as argument expected_url and check it inside. 
Instead of print() use please return "User is redirected to the product page."/"User is not redirected to the product page." based on result.